tests/integration/acp.go

```diff
@@ -357,134 +247,92 @@ type DeleteDocActorRelationship struct {
 	ExpectedError string
 }
 
-func deleteDocActorRelationshipACP(
-	s *state,
-	action DeleteDocActorRelationship,
-) {
-	if action.NodeID.HasValue() {
-		nodeID := action.NodeID.Value()
-		collections := s.collections[nodeID]
+func deleteDocActorRelationshipACP(s *state, action DeleteDocActorRelationship) {
+	processNode := func(nodeID int) {
```

Comment:
Question: why did this action needed to be refactored?

Reply:
there was code duplication in 3-4 places. It is tiresome to keep them in sync. That's why I refactored. This file is shrunk by decent amount of lines now.

Reply:
I am very strongly against this change for 2 reasons:

1) My work resolving: https://github.com/sourcenetwork/defradb/issues/3076 already fixes the duplication issue and the `getNodes` bug. So in addition to being a directly conflict it also doesn't fix the underlying bug.

2) This over abstraction complicates things for no reason, and only applied to acp actions. All other actions now have a different flow and acp actions have a totally different flow. Either do it for all of them or none of them. I do prefer the explicitness of each action the way it is, as it keeps them flexible. I now have to jump to just to `executeACPAction` to know if there are multiple nodes specified does it skip for `SourceHubACPType` or not.

Reply:
> There in no abstraction here. I don't know why you use this word.

A function is an abstraction, I think Shahzad's use of the word is totally fair here.

My (and possibly Shahzad's) primary objection is that you are reworking the `getNodes` stuff in a very ACP specific way that does not solve the general wider problem noted in #3076 - at a time when (I believe) Shahzad is actively working on and is very nearly ready to open up a PR to solve that wider issue in a consistent way for all actions (not just ACP).

You are both solving the same problem at the same time, only Shahzad was working on it first and has solved it across all of our actions, not just ACP.

At the same time your changes have coupled various other, independent ACP action logic into the same helper function (abstraction).  I also disagree with this choice - IMO it is nicer to keep them independent and individually as simple as they can be, especially given that this is essentially untested test code in a very complicated test framework.

Reply:
> you are reworking the getNodes

> You are both solving the same problem

I did not rework anything. Can't you guys see that it's just refactoring? The execution flow is the same. Maybe I'm blind here.

Reply:
You have reworked the code that manages `getNodes` for the ACP actions, something that is actively being worked on by another team member who is looking to solve this awkwardness for all our actions.

We all recognise that there is a problem, I am mostly asking you to be a little patient, tolerate it within the ACP actions for a little longer, and not step on the toes of the developer who has been trying to sort out the wider problem.

By solving the problem for all actions, we should have simpler and more consistent action code, allowing us to maintain test actions a little more easily without worrying about action/action-sub-set specific oddities.

Reply:
This thread is getting a little long, perhaps @islamaliev, @shahzadlone , myself and anyone else interested can have a call in dev-hangout?

Reply:
@AndrewSisley has described all the concerns I had. In summary #3076 resolves the duplication problem and the underlying bug for all actions, and doesn't use another abstraction/function. Was going to open PR over the weekend but was traveling to Dubai, I will have the PR open soon.

Regardless of the decision I don't see how/why this has to be done in this PR, this refactor/abstraction is independent of the Node Identity feature.

Reply:
removed executeAPCAction

Reply:
Seems like you are deciding the keep the `-1` helpers. In that case I would suggest fixing them and making every single action follow that pattern so we can have that `-1` feature to test invalid docID's, collectionIDs etc.

Reply:
> Seems like you are deciding the keep the -1 helpers. 

I'm now really confused. We discussed it yesterday and agreed to keep helpers as is and remove only `executeACPAction`. I didn't decide to keep anything. I just extracted 4 occurrences of this:
```go
var targetIdentity string
if action.TargetIdentity == -1 {
	targetIdentity = ""
} else {
	optionalTargetIdentity := getIdentity(s, i, immutable.Some(action.TargetIdentity))
	if !optionalTargetIdentity.HasValue() {
		require.Fail(s.t, "Expected non-empty target identity, but it was empty.", s.testCase.Description)
	}
	targetIdentity = optionalTargetIdentity.Value().DID
}
```
and other parts into separate functions.
And please leave your comments precisely on a line that you are referring to.

Reply:
This conversation leads nowhere. You merge first your stuff. I will see what I can do later.

Reply:
> We discussed it yesterday and agreed to keep helpers as is and remove only executeACPAction

I also thought we agreed to keep the helpers.  I have a minor preference against the partial refactor, but it is not nearly important enough to drag out the review over, and it shouldn't impact the work Shahzad is currently working on.

Reply:
We did agree to keep the helpers, however all I am saying is to mark that this needs to be done for all actions (as this is a partial refactor). In other words at least have an issue marking we need to do this for all actions (or whatever way we decide to handle in future the invalid/empty cases).

Reply:
Made this: https://github.com/sourcenetwork/defradb/issues/3147

Resolving this thread